### ISSUE TITLE
NonExistentTimeError thrown during DST transition

### ISSUE THREAD

**Issue by babu-upcomer** [mfogel/django-timezone-field#70]
Here is my model field, I need to show all the obsolete timezones as the frontend clients still send the timezone in obsolete formats(Asia/Calcutta instead of Asia/Kokatta). Not sure if that's the reason.

```
TIMEZONE_CHOICES = [(pytz.timezone(tz), tz) for tz in pytz.all_timezones]

timezone = TimeZoneField(choices_display='WITH_GMT_OFFSET', choices=TIMEZONE_CHOICES, blank=True, null=True)
```

Error
```
timezone = TimeZoneField(choices_display='WITH_GMT_OFFSET', choices=TIMEZONE_CHOICES, blank=True, null=True)
  File "timezone_field/fields.py", line 72, in __init__
    choices = with_gmt_offset(values)
  File "timezone_field/choices.py", line 32, in with_gmt_offset
    delta = pytz.timezone(tz_str).utcoffset(now)
  File "pytz/tzinfo.py", line 422, in utcoffset
    dt = self.localize(dt, is_dst)
  File "pytz/tzinfo.py", line 341, in localize
    raise NonExistentTimeError(dt)
```

![Sentry error stack](https://user-images.githubusercontent.com/19949828/111057015-1ebde400-84aa-11eb-9bdc-1c751a4fb10b.png)

**Comment by mfogel** [comment]
Yeah, under the hood this field uses [pytz](https://pythonhosted.org/pytz/) to get metadata about timezones. If pytz doesn't recognize a string as a valid timezone, it will throw one of those `NonExistentTimeError` exceptions. So it looks like that's whats happening here: the obsolete timezones are not being recognized by pytz.

If you have to support the obsolete timezones, perhaps just a standard string field with a `choices` argument might suffice?

**Comment by babu-upcomer** [comment]
@mfogel Thanks for responding! I think the problem isn't with the obsolete timezones even `America/Adak` is in pytz.common_timezones. I guess that it can be fixed with any one of the solutions put out [here](http://www.ilian.io/django-pytz-nonexistenttimeerror-and-ambiguoustimeerror/) and also this exception from pytz is intentional. `is_dst` needs to be sent `True` or `False` based on the exceptions. Since this blocks the initialization of the models and so the gunicorn workers to crash when they restart, I think this needs to be prioritized. I'm sure that every who is using the library now in production might have had bunch of errors for like an hour in the reported time.

Thanks for the work!
- Babu

**Comment by mfogel** [comment]
Ahh, I see what's going on. Thanks @babu-upcomer for the link and more detailed explanation.

Yeah I think for anyone using this field and using the kwarg `choices_display='WITH_GMT_OFFSET'` they probably got this error during the recent north america DST transition.

It looks like Europe will be undergoing a DST transition in a few weeks, I will plan to get this bug fixed and an update pushed to pypi before then.

**Comment by mfogel** [comment]
Fix released as part of [v4.1.2](https://pypi.org/project/django-timezone-field/4.1.2/)
